src/preamble.js

```diff
@@ -1425,11 +1427,13 @@ function setF64(ptr, value) {
 #if USE_PTHREADS
 if (!ENVIRONMENT_IS_PTHREAD) {
 #endif
-  HEAP32[0] = 255;
+  HEAP32[0] = 0x63736d65; /* 'emsc' */
 #if USE_PTHREADS
 }
 #endif
-if (HEAPU8[0] !== 255 || HEAPU8[3] !== 0) throw 'Runtime error: either the current system is not little-endian, or it has corrupted its heap memory area (address zero)!';
+HEAP16[1] = 0x6373;
+if (HEAPU8[2] !== 0x73 || HEAPU8[3] !== 0x63) throw 'Runtime error: expected the system to be little-endian!';
+if (HEAP32[0] !== 0x63736d65) throw 'Runtime error: The application has corrupted its heap memory area (address zero)!';
```

Comment:
could this line be moved 2 higher, before the 16-bit assign? that way we don't ignore corruption of the area the write writes to?